### ISSUE TITLE
Manually mark external study participant status

### ISSUE THREAD

**Issue by jchu10** [lookit/lookit-api#1106]
# TL;DR
For external studies, I want to mark participant responses as "complete" vs. "incomplete".

# Narrative

As a researcher, I have been getting more repeat signups on my external study. I would like to (A) distinguish between who signed up and who actually completed the study, (B) utilize the past participation filter to prevent repeat signups. Currently, the study response page shows all responses as "incomplete" - this includes every user who clicked into my study and got redirected externally, regardless of whether they eventually signed up, and whether they eventually completed the study. 

As a participant, all external studies that I have viewed show up in my "past studies" page, even if I didn't actually complete / schedule an appointment. 

Allowing researchers to manually update participant status would resolve both of these inaccuracies.

# Acceptance Criteria

- [ ] Researchers can mark external study responses as complete/incomplete
- [ ] Researchers can filter by completion status in the "responses" view, and download participant data for just those accounts.
- [ ] Accurate participation status is updated under "past studies" view for participants. 
- [ ] Same downstream effects of completion status as for internal studies, e.g., reminder emails, etc.

# Implementation Notes

One idea is to use something like the consent coding interface. This would be especially nice in combination with allowing external studies to use internal Lookit consent frame.

**Comment by becky-gilbert** [comment]
This issue will be partially addressed here: #1507

Specifically, all studies (including external) will get a new "Response Status" column in the Individual Responses table where researchers can mark them as something like Complete/Incomplete or Pending/Accepted/Rejected (actual labels are TBD). It will also be possible to filter by that "Response Status" value.

I'm not sure yet, but I don't think we'll be implementing the following as part of the above issue:
- download participant data for just those responses/accounts 
    - This could be done either with a "download responses" button on the same page that downloads the IR table results based on any current filters, or as a button on the All Responses page that downloads all responses marked as "Complete"/"Accepted".
- Accurate participation status is updated under "past studies" view for participants.
    - It looks like we don't currently mark the responses in the study history as "Complete" or "Incomplete", so I think the request is to add that information?
- Same downstream effects of completion status as for internal studies, e.g., reminder emails, etc. 
    - I'm not sure what the downstream effects of the completion status are for messaging, because I think most of our filtering for things like emailing relates to whether the person has tried to participate, rather than whether or not they have a complete session. The only thing I can think of is that we could tie the "must have/not have participated" eligibility criteria to this researcher-defined completion status.

**Comment by becky-gilbert** [comment]
This has now been partially implemented in https://github.com/lookit/lookit-api/pull/1534. The new "Session Status" column in the Individual Responses table contains a researcher-editable field in which researchers can mark each response session as completed.

The Session Status drop-down options are: "" (blank), "To schedule", "Scheduled", "Session attended", "Session complete", "Follow up", "Communication complete", "Withdrawn or closed".

This is the same for external and internal studies.

![Image](https://github.com/user-attachments/assets/ff0c608e-89d3-4d46-8c20-5dc0fdcbcbd7)

As I mentioned in the comment above, we have not addressed the other parts of this issue so I will leave it open.

**Comment by mekline** [comment]
@becky-gilbert Break this out into what's not covered by new Indiv. responses page; make new issue

**Comment by becky-gilbert** [comment]
Closing this and breaking out into new issues (links to be added here).
